test/pytest/mockwebserver.py

```diff
@@ -0,0 +1,492 @@
+# This file is part of Cockpit.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# https://github.com/astral-sh/ruff/issues/10980#issuecomment-2219615329
+# ruff: noqa: RUF029
+
+import argparse
+import asyncio
+import binascii
+import contextlib
+import json
+import logging
+import os
+import pwd
+import socket
+import weakref
+from collections.abc import AsyncIterator, Awaitable, Callable, Mapping
+from pathlib import Path
+from typing import ClassVar, NamedTuple, Self
+
+import aiohttp
+from aiohttp import WSCloseCode, web
```

Comment:
As discussed in our 1-1 yesterday and in https://issues.redhat.com/browse/COCKPIT-1238: for the top-level web server we can't use `aiohttp` as that doesn't support HTTP 2/3. The most prominent candidate would be [fastapi](https://fastapi.tiangolo.com/) or the underlying [starlette](https://www.starlette.io/). I wouldn't want our future cockpit-ws package to end up depending on and being written in two different frameworks.

It's probably easier to port it now while it's in its infancy than once we actually start rolling it out.

I'm not yet clear where this is going, though, I'll write something in the top-level review. This might be a follow-up.